lstchain/reco/utils.py

```diff
@@ -611,3 +612,25 @@ def clip_alt(alt):
     to keep astropy happy
     """
     return np.clip(alt, -90.*u.deg, 90.*u.deg)
+
+def get_effecive_time(events):
+    """
+    Calculate the effective observation time of a set 
+    of real data events.
+    Parameters
+    ----------
+    events: pandas DataFrame
+
+    Returns
+    -------
+    teff: float
+    """
+    
+    deltat = np.diff(events['dragon_time'])
+    deltat = deltat[(deltat > 0) & (deltat < 0.002)]
+    rate=1/np.mean(deltat)
+    dead_time = np.amin(deltat)
+    t_elapsed = len(events)/rate * u.s
+    t_eff = t_elapsed/(1+rate*dead_time)
```

Comment:
Are you sure you need to correct for dead time when estimating the total time through the mean of the differences? Shouldn't that be directly an estimate of the effective time?

Reply:
I remember this was discussed already in the notebook where I introduced the function, but now I'm having doubts... perhaps @moralejo can help to clarify this?

Reply:
I made a small toy experiment that I think shows pretty clearly that we already account for the dead time by taking the mean of the detected events if I didn't make a wrong assumption.

![Screenshot_20210225_161500](https://user-images.githubusercontent.com/5488440/109174097-ce921280-7784-11eb-8728-3ec5082fa875.png)

@moralejo ?

Reply:
Sorry, since we get the full observation time back here, we do need to take the dead time into account to get the "live time", so the code should be correct.

Reply:
Thanks for the checking Max,